### ISSUE TITLE
[DOC] null model and ICC in mixed linear model

### ISSUE THREAD

**Issue by StevenLi-DS** [statsmodels/statsmodels#6210]
Could you guys consider adding some details for how to specify a null model and compute ICC? 
https://www.statsmodels.org/devel/examples/notebooks/generated/mixed_lm_example.html

**Comment by StevenLi-DS** [comment]
btw, I think the "Likelihood" in the results summary should be "log-Likelihood", right? Is this intentioned?

![image](https://user-images.githubusercontent.com/33796896/67459909-a8e08080-f607-11e9-82d6-ec0734aca0d8.png)

**Comment by emilmirzayev** [comment]
Yes, `self.llf` is loglikelihood according to the [source code](
https://github.com/statsmodels/statsmodels/blob/5f33aab465f2345483c9d100af58d9f0f8d66504/statsmodels/regression/mixed_linear_model.py#L2508)

**Comment by kshedden** [comment]
I don't have much experience with Jupyter nor a computer that runs it, so I'm not doing anything with notebooks at the moment.

When I need an ICC, I fit a mixed model, then calculate the ICC by hand.  For the growth curve of pigs example (first example in the notebook you linked), the ICC can be estimated as 40.4 / (40.4 + 11.4) ~ 0.78, where 11.4 is the scale parameter (residual variance).

If someone wants to add that discussion to the notebook and PR it, that would be great.

**Comment by StevenLi-DS** [comment]
@kshedden thanks for the fix.

I did some research and I think maybe we should add ICC and LRT test into the function or at least add them in the doc.

```python
import numpy as np
from scipy.stats import chi2
import statsmodels.formula.api as smf
import statsmodels.api as sm

df = sm.datasets.get_rdataset("dietox", "geepack").data

# fit a null-model
formula = 'Weight ~ 1'
null_model = smf.mixedlm(formula, data=df, groups=df.Pig)
results = null_model.fit(reml=False)

def get_icc(results):
    '''get the Intraclass Correlation Coefficient (ICC)'''
    icc = results.cov_re / (results.cov_re + results.scale)
    
    return icc.values[0, 0]


def lr_test(formula, data, groups):
    '''perform likelihood ratio test of random-effects'''
    # fit null model in mixed linear model
    null_model = smf.mixedlm(formula, data=data, groups=groups)\
                    .fit(reml=False)
    # fit OLS model
    ols_model = smf.ols(formula, data=data)\
                   .fit()
    # get the LRT statistic and p-value
    lrt = np.abs(null_model.llf - ols_model.llf) * 2
    p = chi2.sf(lrt, 1)
    
    return (lrt, p)

# get the ICC and LRT results
icc = get_icc(results)
lrt, p = lr_test(formula, data=df, groups='Pig')

print(f'ICC = {icc:.4f}')
print(f'The LRT statistic: {lrt:.4f} (p = {p:.5})')
```
```
ICC = 0.0228
The LRT statistic: 4.9646 (p = 0.025871)
```
The results can be verified in r:
```r
library(tidyverse)
library(lmerTest)
library(performance)

df = read.csv('pigs.csv', fileEncoding='UTF-8-BOM')

null_model = lmer('Weight~ 1 + (1|Pig)', data=df, REML=FALSE)

model_icc = model_performance(null_model)$ICC
lrt = ranova(null_model)

sprintf('ICC = %.4f', model_icc)
sprintf('p-value for LRT: %e', lrt$Pr[2])
```

**Comment by kshedden** [comment]
This could be helpful for some users.  The implementation you gave only works if there is a single random effect that corresponds to random intercepts for a collection of groups.  There would need to be some way to check for this and return a warning or error otherwise.

The standard LRT isn't correct here because the hypothesis space is a closed set and the null is on the boundary.  There is a vast literature on this topic, which I don't know very well.  There are procedures by Satterthwaite and Kenward-Roger that are widely used.

**Comment by StevenLi-DS** [comment]
@kshedden thanks for the info. I couldn't find the doc for that function right now but I simply replicated the function `rand` or `ranova` in `lmerTest` library. Based on page 20 of their [doc], it uses Satterthwaite’s approximation. The p-value I got indeed equals the p-value I got from their package. Maybe I missed something?

https://stackoverflow.com/questions/38248595/likelihood-ratio-test-in-python
